src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -638,6 +638,38 @@ public static IOperation WalkDownConversion(this IOperation operation)
 
             return thrownObject?.Type;
         }
+
+        public static bool HasAnyExplicitDescendant(this IOperation operation, Func<IOperation, bool>? descendIntoOperation = null)
+        {
+            var stack = ArrayBuilder<IEnumerator<IOperation>>.GetInstance();
+            stack.Add(operation.Children.GetEnumerator());
+
+            while (stack.Any())
+            {
+                var enumerator = stack.Last();
+                stack.RemoveLast();
+                if (enumerator.MoveNext())
+                {
+                    var current = enumerator.Current;
+                    stack.Add(enumerator);
+
+                    if (current != null &&
+                        (descendIntoOperation == null || descendIntoOperation(current)))
+                    {
+                        if (!current.IsImplicit &&
+                            // This prevents non explicit operations like expression to be considered as ok
```

Comment:
Nit: consider moving the comment outside the if expression. Can do it as a follow-up PR or part of any other PR in future.